libs/community/langchain_community/retrievers/azure_ai_search.py

```diff
@@ -79,10 +86,17 @@ def _build_search_url(self, query: str) -> str:
 
     @property
     def _headers(self) -> Dict[str, str]:
-        return {
-            "Content-Type": "application/json",
-            "api-key": self.api_key,
-        }
+        if self.azure_ad_token_provider:
+            token = "Bearer " + f"{self.azure_ad_token_provider()}"
+            return {
+                "Content-Type": "application/json",
```

Comment:
To avoid repeating the Content-Type, you could do something like:

headers = {
  "Content-Type": "application/json"
}

if self.azure_ad_token_provider:
   headers["authorization"] = ...
else:
   headers["api-key"] = ...
return headers


Kind of nice especially if headers increase.

Reply:
Sure! Will update

Reply:
Done!